### DISCUSSION TITLE
Material You Layout

### DISCUSSION THREAD

**Discussion by greghesp** [cagnulein/qdomyos-zwift#749] [Ideas]
Just something I've been playing around with.  Only a rough first thought, but be interested to hear peoples thoughts?

![iPhone 12 Render](https://user-images.githubusercontent.com/768908/160933339-ad8d2526-80b0-40c2-8eb9-2fd0604eb9eb.png)
![iPhone 12 Render (1)](https://user-images.githubusercontent.com/768908/160933342-2e2a71f7-0b92-4955-8f22-f659c3bc36d3.png)

**Comment by cagnulein** [comment]
I love them! please share it on facebook group!

**Reply by greghesp** [reply]
Will do!

**Comment by greghesp** [comment]
Bit of a tweak

![iPhone 12 Render](https://user-images.githubusercontent.com/768908/161104115-1ef9414c-6d9d-4333-939d-4fdf37a69842.png)

**Reply by cagnulein** [reply]
very clean! keep in mind that the users like to have every aspect of the UI customizable as you saw also on the facebook group ;)

**Comment by tylerjames** [comment]
Is this a native UI for Android?

I'm kind of interested in making one for iOS as well. My needs for this are minimal right now so my goals would mainly be simplicity and a native UI feel.

Is that doable?

This looks pretty slick though, I like it.

**Comment by cagnulein** [comment]
QZ is already available for ios, it's in the store!


Il giorno gio 6 apr 2023 alle 18:19 tylerjames ***@***.***>
ha scritto:

> Is this a native UI for Android?
>
> I'm kind of interested in making one for iOS as well. My needs for this
> are minimal right now so my goals would mainly be simplicity and a native
> UI feel.
>
> Is that doable?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/749#discussioncomment-5545485>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWE44EWXQA3SOR7CL3TW73UJTANCNFSM5SDG7OOA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Comment by tylerjames** [comment]
Yeah I was just thinking of something where the UI is built in SwiftUI or UIKit instead of QT

**Reply by cagnulein** [reply]
@tylerjames actually my idea (and i already started) is to use the integrated web server on qz to make a web app.

but if you want to use swift ui will be nice!

you can start a branch here!

**Reply by tylerjames** [reply]
Yeah if I ever end up finding the time I will. Was thinking to just try a really stripped down version for my own small use case first and see how it goes.

There's a lot to unpack here though! I don't know how you ever managed to figure this out but well done

**Reply by cagnulein** [reply]
i lost my last 3 years of my life on this :) if you need any help let me know

**Reply by tylerjames** [reply]
Haha, all because you couldn't stomach paying for Peloton!

**Reply by cagnulein** [reply]
actually peloton is not even available here in italy :)  

no i started because i hate the closed solutions!

**Reply by tylerjames** [reply]
I'm brand new to this whole space (got introduced to Zone 2 training by Peter Attia) and couldn't stomach the cost of a Peloton. Ended up getting an Echelon for less than half the cost and your app was a big determining factor on that decision because it meant I wouldn't be locked into their shitty app either. I just want to be able to get the metrics off my bike and track my workouts. Maybe I'll eventually try Zwift or the Peloton app or whatever but for now I just want to get my 3-4 hours of Zone 2 per week and maybe one Zone 5 workout in. 

The app comes universally recommended and even my wife's coworkers already knew about it and used it so it was an easy purchase. Thanks for all the work you've put into it. 

As for Swift/SwiftUI we'll see how that goes. I've never tried to integrate a pure Swift app with a largely C++ code base before. I'm not even quite sure what it would take to get a "hello world" where I've launched the Swift app and maybe connected to my bike. I'll try poking around with it though and let you know if I get anywhere

**Reply by cagnulein** [reply]
actually i already integrated the qz c++ code in a library (for example kinomap is doing this in their code with the qz library). so it's doable and also easy to do :)

are you a swift developer as your main job?

**Reply by tylerjames** [reply]
Yeah I am, but it's been roughly two decades since I've touched C++ and I've never made an app that relied on it heavily.

**Reply by cagnulein** [reply]
ok :) let me know :)
